core/io/xml_parser.cpp

```diff
@@ -194,14 +225,43 @@ void XMLParser::_parse_comment() {
 #endif
 }
 
-void XMLParser::_parse_opening_xml_element() {
+Error XMLParser::_parse_opening_xml_element() {
+	// check if the first character is a digit
+	if (isdigit(*P)) {
+		ERR_FAIL_V_MSG(ERR_INVALID_DATA, "Invalid tag name, a tag cannot begin with a number.");
+	}
```

Comment:
You should use the dedicated checks here:
```cpp
ERR_FAIL_COND_V_MSG(isdigit(*P), ERR_INVALID_DATA, "Invalid tag name, a tag cannot begin with a number.");
```

Reply:
Got it, thanks!